Update docs to include additional dependencies and clear up info regarding docker (#523)
FuelLabs/fuel-indexer: docs/README.md

- We use Docker to produce reproducible environments for users that may be concerned with installing components with large sets of dependencies (e.g. Postgres).
- Docker can be downloaded [here](https://docs.docker.com/engine/install/).

> ### `postgresSQL` and `SQLite`

Reviewer: @bitzoic 
- I suggest we:
  - Add `PostgreSQL` and `SQlite` sub-headings under `For Users > Dependencies`.
  - Then link those sub-headings here, in `PostgreSQL` and `SQLite` sections
  - Then for now, we can just link to the download page (indexer team can follow up with more later)

Thoughts?

> - We use PostgresSQL and SQLite to manage our database in the following quickstart.

Reviewer: @bitzoic 
- So Postgres and SQLite are actually not required for the Quickstart (because we use Docker), which is why we forgot to add them to this "Users" section of the README.
  - But they _should_ be in the "Users" section of the README (see comment above ☝️) - so good catch here.

Author: Ah, I only now saw that this is a "For Users" section. As per my other comment, since it's specifically a quickstart and not required I'll remove this.

- Postgres can be downloaded [here](https://www.postgresql.org/download/)
- SQLite can be downloaded [here](https://www.sqlite.org/download.html)

> ### `wasm`

Reviewer: Great catch!

> - Two additonal rust-based components required are wasm-snip and wasm32.

Reviewer: ```suggestion
Two additonal cargo components will be required to build your indexers: `wasm-snip` and the `wasm32-unknown-unknown` target.
```

> - Run the following command to install wasm-snip:

Reviewer: ```suggestion
To install `wasm-snip`:
```

```bash
cargo install wasm-snip
```

> - Run the following command to install wasm32:

Reviewer: ```suggestion
To install the `wasm32-unknown-unknown` target via `rustup`:
```

+--------+------------------------+------------------------------------------------------------------+
```

> If there are any missing components please install them in the [Dependencies](#dependencies) section. 

Reviewer: So some of these components being missing is OK for the Quickstart (e.g., `psql`, `fuel-core`, etc), but some are required (e.g., `docker`, `fuelup`) - so we (indexer) should make this clearer in a follow up PR.

Author: I will go ahead and remove this for now then but I think it would be valuable to show the output in the `2.1 Check for components` section only showing what is required rather than all components installed in that follow up PR👍

#### 2.3.1 Starting an indexer service

- To start an indexer service, we'll be spinning up Postgres and Fuel indexer containers via `docker compose`. Our indexer service will connect to Fuel's `beta-2` network so that we can index blocks and transactions from an _actual_ Fuel node. We'll use the `docker-compose.yaml` file below, and spinning everything up with `docker compose up`.

> IMPORTANT: Ensure that any local Postgres instance that is running on port `5432` is stopped.

> 

Reviewer: 👌

      - postgres
```

> Then run `docker compose up` to spin up Postgres and Fuel indexer containers:

Reviewer: 👍
